### PR TITLE
MNT: Use f-strings where possible

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Use `.format()` where f-strings would be unreadable.

### CODE REVIEW

File: smriprep/workflows/anatomical.py [hunk 1]
```diff
@@ -222,15 +222,11 @@ def init_anat_preproc_wf(
...
+    desc = f"""
 Anatomical data preprocessing
 
-: """
-    desc += """\
-A total of {num_t1w} T1-weighted (T1w) images were found within the input
-BIDS dataset.""".format(
-        num_t1w=num_t1w
-    )
+: A total of {num_t1w} T1-weighted (T1w) images were found within the input
```
Comment (DimitriPapadopoulos) [inline smriprep/workflows/anatomical.py:228]: The resulting string `desc` is:
```
Anatomical data preprocessing

: A total of num_t1w T1-weighted (T1w) images were found within the input
BIDS dataset.
```
Is this really what we want?

**Comment by effigies** [comment]
I'd rather not merge cosmetic changes in before we merge [next](https://github.com/nipreps/smriprep/tree/next) into master.

Also, `%` is not deprecated functionality, so let's not "upgrade" to `.format()` just because it's newer.

File: smriprep/workflows/anatomical.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply smriprep/workflows/anatomical.py:228]: @effigies What about this one? Was on purpose?

Comment (effigies) [reply smriprep/workflows/anatomical.py:228]: I think this is a misreading. If you check https://output.circle-artifacts.com/output/job/c1a05c82-b211-4baa-88e9-2b93fcd61dbc/artifacts/0/bids-run/sub-01.html#boilerplate, we do interpolate the number.

Comment (DimitriPapadopoulos) [reply smriprep/workflows/anatomical.py:228]: I was referring to the leading **': '** at the start of the paragraph. The Mardown outptut looks like this:
```txt
Anatomical data preprocessing

: A total of 1 T1-weighted (T1w) images were found within the input
BIDS dataset.The T1-weighted (T1w) image was corrected for intensity non-uniformity (INU)
```
and as far as I know **': '** does not have a special meaning, and is rendered:
```markdown
Anatomical data preprocessing

: A total of 1 T1-weighted (T1w) images were found within the input
BIDS dataset.The T1-weighted (T1w) image was corrected for intensity non-uniformity (INU)
```

Comment (effigies) [reply smriprep/workflows/anatomical.py:228]: I think technically it's using [https://garrettgman.github.io/rmarkdown/authoring_pandoc_markdown.html#definition-lists](definition lists) to make "Anatomical data processing" into almost-but-not-quite a section heading.

Comment (DimitriPapadopoulos) [reply smriprep/workflows/anatomical.py:228]: That was the https://markdownlivepreview.com/ renderer that doesn't understand this Pandoc extension.

Using https://dillinger.io, the Markdown output looks slightly better, but still not perfect (the newline is part of the rendereding):

markdown

Is the Markdown output just for conversion with Pandoc, or is it supposed to be usable as such?

**Comment by DimitriPapadopoulos** [comment]
The idea is that f-strings are faster, not newer.

**Comment by effigies** [comment]
To be clear, I'm fine upgrading to f-strings where it works and is clean. But when we're using `%` instead of `.format()`, it's usually because it's a delayed interpolation and we decided `%` was the cleaner of the two options.

**Comment by DimitriPapadopoulos** [comment]
I understand. In any case, isn't it better to wait for next to be merged into master?

**Pull request closed without merging**